src/NATS.Client.Core/NatsMsg.cs

```diff
@@ -259,3 +299,29 @@ private void CheckReplyPreconditions()
         }
     }
 }
+
+public class NatsMsgError
```

Comment:
Seems a little redundant to have an error class that just ends up holding a couple different exceptions.  Would it be better to :

```csharp
public class NatsDeserializeException : NatsException {
  public NatsDeserializeException(inner): NatsException("Exception during deserilaiztion", inner)

  public byte[] Data { get; internal set; }

}

public class NatsHeaderParseException : NatsException {
  public NatsHeaderParseException(inner): NatsException("Exception parsing headers", inner)

  public byte[] Data { get; internal set; }
}
```

Reply:
I like that. But we 'might' have two exceptions at the same time and we'd only have one property on the message i.e. `msg.Error`.

Should we not handle header parser exceptions here?

They are unlikely to happen (only on some client producing malform headers) which would end up in log / on sub dispose.

Reply:
Would the 2 exceptions be a Header Parser exception and a De-serialize exception?  Or a Serialization exception and some other exception?

If it is Header Parser exception could we just hit that one stop, not even attempt the de-serialize?

Reply:
yep it's Header Parser exception and a De-serialize exception.

sounds good!